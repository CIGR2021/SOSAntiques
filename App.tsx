/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import './global.css';
import { GluestackUIProvider } from '@/components/ui/gluestack-ui-provider';
import { NavigationContainer } from '@react-navigation/native';
import Routes from './src/routes';
import { AppProvider } from './src/context/AppProvider';
import { Box } from './components/ui/box';
import { I18nextProvider } from 'react-i18next';
import i18n from './src/config/i18n';

const App = () => {
  return (
    <Box className="flex-1" testID="app-container">
      <I18nextProvider i18n={i18n}>
        <GluestackUIProvider>
          <AppProvider>
            <NavigationContainer>
              <Routes />
            </NavigationContainer>
          </AppProvider>
        </GluestackUIProvider>
      </I18nextProvider>
    </Box>
  );
};

export default App;
