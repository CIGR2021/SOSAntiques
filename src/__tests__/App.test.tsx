import React from 'react';
import { render } from '@testing-library/react-native';
import App from '@SOSAntiques/App';

// Mock do arquivo CSS
jest.mock('./global.css', () => ({}));

// Mock dos componentes necessÃ¡rios
jest.mock('@/components/ui/gluestack-ui-provider', () => ({
  GluestackUIProvider: ({ children }: { children: React.ReactNode }) =>
    children,
}));

jest.mock('@/src/routes', () => {
  const { Text } = require('react-native'); // Importar dentro do mock
  return () => <Text testID="mocked-routes">Mocked Routes</Text>;
});

jest.mock('@/src/context/AppProvider', () => ({
  AppProvider: ({ children }: { children: React.ReactNode }) => {
    const MockedContext = require('react').createContext({ isLoggedIn: false });
    return (
      <MockedContext.Provider value={{ isLoggedIn: true }}>
        {children}
      </MockedContext.Provider>
    );
  },
}));

jest.mock('react-native/Libraries/Utilities/useWindowDimensions', () => {
  return jest.fn(() => ({
    width: 360,
    height: 640,
    scale: 2,
    fontScale: 1,
  }));
});

describe('App', () => {
  it('renderiza sem travar', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('app-container')).toBeTruthy();
    expect(getByTestId('mocked-routes')).toBeTruthy(); // Verificando o mock de Routes
  });
  it('JSX retornado', () => {
    const tree = render(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
