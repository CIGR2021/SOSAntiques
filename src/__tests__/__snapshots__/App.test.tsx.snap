// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App JSX retornado 1`] = `
<View
  className="flex-1"
  testID="app-container"
>
  <Text
    testID="mocked-routes"
  >
    Mocked Routes
  </Text>
</View>
`;

exports[`App matches snapshot 1`] = `
<View
  className="flex-1"
  testID="app-container"
>
  <Text
    testID="mocked-routes"
  >
    Mocked Routes
  </Text>
</View>
`;
